tests/kernel/critical/src/Makefile

```diff
@@ -1,4 +1,3 @@
 ccflags-y += -I${ZEPHYR_BASE}/tests/include
 include $(ZEPHYR_BASE)/tests/Makefile.test
-
-obj-y = critical.o
+obj-y = main.o test_critical_context.o
```

Comment:
@Vudentz  Some test cases have their own name in src/ directory like critical test case has file name critical.c , mslab test case has file name slab.c. All these looks different and messy. If we can have one main file and other file (whatever name is) , all the test case would looks same and easy to understand structure for any one.